### ISSUE TITLE
Should patterns like `ifelse(x, 1, 0)` be linted in `redundant_ifelse_linter()`? 

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1810]
## Description

> `ifelse(x, 1, 0)` is a false positive.
>
> It's a perfectly valid, logical way of expressing this (and arguably even superior to `as.numeric(x)` because it describes the intended semantics rather than implicitly relying on an implementation detail).

Originally reported by @klmr. 

---

## Example

``` r
library(lintr)

lint(
  text = "ifelse(x, 1, 0)",
  linters = redundant_ifelse_linter()
)
#> <text>:1:1: warning: [redundant_ifelse_linter] Prefer as.numeric(x) to ifelse(x, 1, 0) if really needed.
#> ifelse(x, 1, 0)
#> ^~~~~~~~~~~~~~~
```

<sup>Created on 2022-12-07 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
> implicitly relying on an implementation detail

strongly disagree with this. R is a _statistical language_ first and foremost and therefore TRUE=1, FALSE=0 is extremely fundamental to the language.

Nevertheless, I can see the appeal of allowing this usage with an option (I'd prefer to leave the option off by default, i.e. the current behavior).

**Comment by MichaelChirico** [comment]
ah! forgot we'd already added that.

happy to close this then if we are aligned on the current default.

**Comment by IndrajeetPatil** [comment]
Yeah, I don't think there is much to do here, since we aren't planning to change the default behaviour.
